### PR TITLE
CLAUDE.md @-import include model — vox reference implementation (vox-ys4z)

### PR DESCRIPTION
**Pull request by jmf-pobox**
## Summary

Reference implementation of the CLAUDE.md `@`-import include model (vox-ys4z),
global scope, Option A (vox self-registers). Retires the "inject a block into
`~/.claude/CLAUDE.md`" pattern in favor of a tool-owned doc that CLAUDE.md
`@`-imports.

- **`vox install`** writes an agent-facing usage guide to
  `~/.punt-labs/vox/CLAUDE.md` (how to drive the `mic` MCP tools + `/music`,
  `/vibe`, `/vox`, `/unmute`, `/mute`, `/recap`) and self-registers the line
  `@~/.punt-labs/vox/CLAUDE.md` in `~/.claude/CLAUDE.md`.
- The line lives inside a **shared** marker pair
  `<!-- punt:mandatory-reading -->` … `<!-- /punt:mandatory-reading -->`, headed
  `## Tool Guidance (auto-loaded)`, so a future punt-owned multi-tool reconcile
  supersedes this cleanly.
- The installer owns + rewrites the guide every run — single source of truth,
  never stale.
- `vox uninstall` deletes the guide and prunes the import line.

New module `claude_md.py` (`GlobalClaudeImports` + `VoxGuidance`). The curl
`install.sh` path installs the plugin directly (never via `vox install`), so a
hidden `vox register-guidance [--remove]` command is called from `install.sh`
after the plugin install.

## Why

Per-project CLAUDE.md edits do not scale and go stale. A single global
`@`-import, resolved by Claude Code into every session regardless of CWD, gives
agents vox usage guidance in any project with one install-time registration.

## Review findings addressed

- **F1c (load-bearing, verified not assumed)** — see empirical result below.
- **F2** — the section + `@`-lines are top-level markdown, never inside a code
  fence (fenced imports are not resolved). Test: `test_registered_import_is_top_level_not_in_fence`.
- **F4** — reconcile is deterministic (import lines sorted), corruption-safe (a
  lone open marker or duplicate section is repaired to one canonical section, never
  appended), and no-op when unchanged (rendered text is compared; the shared
  global file is written only if different — no mtime churn). Tests:
  `test_register_is_idempotent_no_op_unchanged`, `test_lone_open_marker_is_repaired_not_duplicated`,
  `test_stray_close_marker_is_removed`, `test_duplicate_sections_collapse_to_one`.
- **F5** — uninstall deletes the guide and prunes the import.
- **F8 (migration, no shim)** — vox never injected a legacy
  `<!-- vox:capabilities -->` block (grep across `src/`, `hooks/`, `scripts/`,
  `commands/`, `install.sh` → none), so there is nothing to remove. The live
  `<!-- quarry:capabilities -->` block in `~/.claude/CLAUDE.md` is left untouched
  (separate/tracked). Verified: install→uninstall restores the file byte-identical,
  quarry block intact.
- **F9** — header is `## Tool Guidance (auto-loaded)`, not "Mandatory Reading …
  Read before writing code".
- **F12** — the global guide is at `~/.punt-labs/vox/CLAUDE.md` (home), distinct
  from the repo-local `.punt-labs/vox/vox.md` config; no collision (different dir,
  different filename).

## F1c — empirical load verification

**Result: LOADED. Claude Code resolves `@~/.punt-labs/vox/CLAUDE.md` into a real
session's context, and `@`-imports resolve `~`.**

Method: registered the real import into `~/.claude/CLAUDE.md`, planted a unique
canary token (`VOXIMPORT-9Q2X7K`) in the guide, then ran `claude -p` (v2.1.202)
from multiple working directories asking it to repeat the canary line.

| CWD | Result |
|-----|--------|
| `/tmp` | canary returned |
| `$HOME` (`/Users/jfreeman`) | canary returned |
| `/tmp/vox-f1c-nonpunt` (arbitrary non-punt) | canary returned |
| `/Users/jfreeman/Coding` (outside punt-labs) | canary returned |

Caveats:
- One-shot headless answers vary in *phrasing*: a "match the pattern
  `VOXIMPORT-XXXXXXX`" prompt sometimes returned `NONE` while a "repeat the line
  containing 'canary token'" prompt returned the token from the same CWD. The
  context was loaded in every case; the NONE responses were answer-formatting
  variance, not a load failure. Re-probing with the direct-line prompt returned
  the canary from all four CWDs.
- Verification used `env -u ANTHROPIC_API_KEY -u ANTHROPIC_AUTH_TOKEN claude -p`
  to route through the interactive auth; an invalid `ANTHROPIC_API_KEY` in the
  environment otherwise blocked the probe.

The operator's `~/.claude/CLAUDE.md` was restored byte-identical to its
pre-verification state after testing (install→uninstall round-trips cleanly).

## Manual-verification playbook

Run through the installed artifact (`uv tool install --force dist/*.whl`):

```
vox register-guidance            # writes guide + registers import (marker count: 2)
vox register-guidance            # 2nd run → "already registered", no rewrite
vox register-guidance --remove   # deletes guide + prunes import
```

Observed: first run wrote `~/.punt-labs/vox/CLAUDE.md` and the section; second run
was a no-op ("already registered"); `--remove` deleted the guide and pruned the
section, leaving `~/.claude/CLAUDE.md` byte-identical to its original backup.
Wheel packaging confirmed to include `assets/global-guidance.md`.

**Operator confirmation requested:** the F1c table above is the load-bearing
claim — please confirm the empirical result is acceptable as the standard's
Known-Use proof.

## Out of scope (companion follow-up)

Finalizing the punt-kit `standards/claude-md-import-includes` branch (citing vox
as the real Known Use) is a separate step after this proves the mechanics, per
the bead.

---

> [!NOTE]
> **Medium Risk**
> The PR mutates the user's global `~/.claude/CLAUDE.md` and home state; behavior is heavily tested but incorrect reconcile could still affect unrelated content outside the managed markers.
> 
> **Overview**
> Adds a **global agent usage guide** that Claude Code loads via a top-level `@` import in `~/.claude/CLAUDE.md`, instead of per-project edits or injected capability blocks.
> 
> **`vox install`** gains a third step: write `~/.punt-labs/vox/CLAUDE.md` from bundled `global-guidance.md` and register `@~/.punt-labs/vox/CLAUDE.md` inside a shared `<!-- punt:mandatory-reading -->` section (`## Tool Guidance (auto-loaded)`). **`vox uninstall`** always tears down the guide and prunes that import (even if the plugin step fails), with non-zero exit if teardown fails.
> 
> New **`GlobalClaudeImports`** (`claude_md.py`) reconciles the managed section: sorted imports, repairs duplicate/lone markers, fence-aware parsing, atomic writes (symlink-safe, preserves line endings/mode), write-only-on-change. **`VoxGuidance`** (`guidance.py`) owns install/uninstall and rollback if registration fails.
> 
> **`install.sh`** probes and calls hidden **`vox register-guidance`** when the binary supports it (curl installs that skip `vox install`). Docs/changelog updated; broad tests cover reconciler edge cases and CLI home redirection.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c12b42953ddf1972cca2f5d7addb70c478452364. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/punt_vox/claude_md.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+        """Return whether *stripped* is a bare closing fence for the open char.
+
+        A closing fence is a run of the open character (three or more) with no
+        trailing info string; ```` ```python ```` opens but never closes.
+        """
+        char = self._char
+        if char is None:
+            return False
+        return stripped.startswith(char * 3) and stripped == char * len(stripped)
+
```
[the remark this answers is not in the document]
Comment (jmf-pobox) [reply src/punt_vox/claude_md.py:114]: Acknowledged as a known, non-blocking markdown edge, not fixing now. Trigger is not realistic: it requires a literal `<!-- punt:mandatory-reading -->` marker placed *inside* a mismatched-width fenced block (opened with 4+ backticks, closed by a shorter run) in the user`s own `~/.claude/CLAUDE.md`. The fence handling here is already defense-in-depth on a punt-specific marker string that essentially never appears in a user file except where punt writes it at column 0; fence-*width* tracking is defense-in-depth on that, for a sub-case with no real-world occurrence. If it is ever hit, the fix is a 3-line width check in `_Fence._closes`. Filing as a follow-up rather than extending this reference-impl PR further.

File: src/punt_vox/claude_md.py [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+        imports: set[str] = set()
+        inside = False
+        fence = _Fence()
+        for line in text.splitlines(keepends=True):
+            stripped = line.strip()
+            was_fenced = fence.inside
+            if fence.feed(stripped) or was_fenced:
+                # Fence delimiter or fenced content: literal, never a marker.
+                kept.append(line)
+                continue
```
[the remark this answers is not in the document]
Comment (jmf-pobox) [reply src/punt_vox/claude_md.py:359]: Non-blocking markdown edge, not fixing. A 4-space-indented triple-backtick is markdown indented-code-block content, not a fence opener; treating it as a fence only matters if a punt marker also sits inside that indented region of the user's ~/.claude/CLAUDE.md, which does not occur in practice. Marker detection is already column-0-only, so this is defense-in-depth on the fence heuristic. Filed as a follow-up if ever hit.

File: src/punt_vox/claude_md.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+        is forced to 0644 (``rw-r--r--``) -- a predictable, sane default for a
+        config file -- rather than the 0600 ``mkstemp`` gives the temp. 0644 is
+        a deliberate constant, *not* the umask-derived mode a plain ``open()``
+        would produce (``0666 & ~umask``); a restrictive umask would otherwise
+        make the created file's mode vary by environment. Called before the
+        rename, while the target (if any) still exists.
+        """
+        if target.is_file():
+            return stat.S_IMODE(target.stat().st_mode)
+        return 0o644
```
[the remark this answers is not in the document]
Comment (jmf-pobox) [reply src/punt_vox/claude_md.py:312]: Intentional and already documented. Forcing 0644 for a brand-new ~/.claude/CLAUDE.md is a deliberate, predictable default for a config file (the _replacement_mode docstring states this explicitly, not umask-derived). Existing files keep their own mode. Not changing to a umask-derived mode — predictability is the design choice here. Resolving.
